synapse/appservice/api.py

```diff
@@ -201,7 +202,13 @@ async def _get() -> Optional[JsonDict]:
         key = (service.id, protocol)
         return await self.protocol_meta_cache.wrap(key, _get)
 
-    async def push_bulk(self, service, events, txn_id=None):
+    async def push_bulk(
```

Comment:
While we're here, can you add a docstring here? IN particular what is `ephemeral` supposed to be? I'm guessing it is actually `JsonDict`, not `Any`?